packages/x-components/src/x-modules/facets-next/store/__tests__/getters.spec.ts

```diff
@@ -0,0 +1,194 @@
+import { Filter } from '@empathyco/x-types-next';
+import { createLocalVue } from '@vue/test-utils';
+import Vuex, { Store } from 'vuex';
+import {
+  createNextEditableNumberRangeFilter,
+  createNextHierarchicalFilter,
+  createNextNumberRangeFilter,
+  createRawFilter,
+  createNextSimpleFilter
+} from '../../../../__stubs__/filters-stubs.factory';
+import { ActionsDictionary, XActionContext } from '../../../../store/actions.types';
+import { MutationsDictionary } from '../../../../store/mutations.types';
+import { arrayToObject } from '../../../../utils/array';
+import { Dictionary } from '../../../../utils/types';
+import { facetsNextXStoreModule } from '../module';
+import {
+  FacetsNextActions,
+  FacetsNextGetters,
+  FacetsNextMutations,
+  FacetsNextState
+} from '../types';
+import { resetFacetsStateWith } from './utils';
+
+/** Type safe single module store. */
+type SafeStore<
+  State extends Dictionary,
+  Getters extends Dictionary,
+  Mutations extends MutationsDictionary<Mutations>,
+  Actions extends ActionsDictionary<Actions>
+> = Omit<XActionContext<State, Getters, Mutations, Actions>, 'rootGetters' | 'rootState'>;
+
+describe('testing facets module getters', () => {
+  function createFacetsStore(
+    filters: Filter[]
+  ): SafeStore<FacetsNextState, FacetsNextGetters, FacetsNextMutations, FacetsNextActions> {
+    const localVue = createLocalVue();
+    localVue.use(Vuex);
+    const store = new Store<FacetsNextState>(facetsNextXStoreModule as any);
+    resetFacetsStateWith(store, { filters: arrayToObject(filters, 'id') });
+    return store;
+  }
+
+  describe(`selected filters getter`, () => {
+    it('returns an empty array if there are no filters', () => {
+      const store = createFacetsStore([]);
+
+      expect(store.getters.selectedFilters).toEqual([]);
+    });
+
+    it('returns an array containing the selected filters', () => {
+      const store = createFacetsStore([
+        createNextSimpleFilter('color', 'Red', false),
+        createNextSimpleFilter('color', 'Blue', true),
+        createNextHierarchicalFilter('category', 'Summer', false),
+        createNextHierarchicalFilter('category', 'Shorts', true),
+        createNextNumberRangeFilter('price', { min: 0, max: 25 }, false),
+        createNextNumberRangeFilter('price', { min: 25, max: 50 }, true),
+        createNextEditableNumberRangeFilter('age', { min: null, max: 5 }),
+        createNextEditableNumberRangeFilter('size', { min: null, max: null }),
+        // Raw filters are always selected, but shouldn't be returned in this getter.
+        createRawFilter('size:xl')
+      ]);
+
+      //expect(store.getters.selectedFilters).toHaveLength(4);
+      expect(store.getters.selectedFilters).toEqual(
+        expect.arrayContaining([
+          store.state.filters['color:Blue'],
```

Comment:
I'm a little bit concerned about having to know how ids are created in the stubs. Is this too much knowledge for this test? The other way of creating the filters in variables and then passing them to the store was too verbose I think.

Reply:
I agree. Maybe we can put it in the `createNextSimpleFilter` doc. Something like:

```
/**
 * Creates a {@link @empathyco/x-types-next#SimpleFilter | SimpleFilter}.
 *
 * @remarks The created simple filter id is going to be the following: <facetId>:<label>.
 *
 * @param facetId - The facet id this filter belongs to.
 * @param label - The text that this filter should display.
 * @param selected - True when the filter is checked, false otherwise.
 * @returns A stub for a {@link @empathyco/x-types-next#SimpleFilter | SimpleFilter}.
 */
export function createNextSimpleFilter(
...
```